### DISCUSSION TITLE
go sdk

### DISCUSSION THREAD

**Discussion by chensanle** [milvus-io/milvus#15557] [Q&A and General discussion]
why I used the python benchmark script can get result just need 1 millisecond in a single request，but use go-sdk examples (2.0.0) need 200-300 millisecond?

**Comment by xiaofan-luan** [comment]
I guess it might be the consistency issue.
We changed to python SDK to bounded staleness but go SDK still use the old strong consistency level.
For strong consistency we need to wait until write processed by the querynode then search start.
@congqixia do we have plan change the default consistency level?

**Reply by chensanle** [reply]
I did the performence test with only one thread , and without any other operations(like insert)，even this can affect performance? （thanks for your answer ❤️）

**Reply by xiaofan-luan** [reply]
We have a timetick mechanism to illustrate that the data before certain time is arrived.
Even if you didn't insert the timetick is issued by rootcoord every 200 ms, that's why querynode need to wait for the tt to arrive before search.

Easiest way to solve this problem would be specifying ganruatee timestamp to 1 when you do search, you can try and see if that's the problem

**Reply by chensanle** [reply]
When specifying ganruatee timestamp to 1, the problem is solved. Unfortunately, it seems that parameter can only be set inside the milvus-sdk-go pkg.

**Reply by congqixia** [reply]
@chensanle Thanks for pointing out this problem. Currently go sdk cannot change consistency level or guarantee timestamp. This feature shall be added in next version, which I think will be release this week.

**Reply by chensanle** [reply]
Wait in hope for the new release, thanks.
